Dev update

SirKnightDragoon opened the pull request.

ivanpopelyshev committed:
fix MovieClip texture update issue (#2653)

themoonrat committed:
Some display object properties weren't showing up in jsdoc (#2655)

* Added @memberof in documentation to have some getters & setters show up in jsdoc

* And hiding an internal member

ivanpopelyshev committed:
Mesh tint, we want it! (#2656)

themoonrat committed:
Easy addition of gradients on text (#2528)

* Allow the easy created of gradients on text just by supplying an array of colours for fill rathar than a string

* removing built files

* Moved generating of the gradient into it's own function

* Fixed function typo

* Fix minor merge error

* getColor can now parse an array of values

* Updated jsdoc to show you can give an array of numbers for fill colour

themoonrat committed:
no-one likes a missed semi-colon :/

englercj committed:
Merge pull request #2666 from themoonrat/docs-fixes-again

No-one likes a missed semi-colon :/

ivanpopelyshev committed:
assignment by value for position/scale/skew

ivanpopelyshev committed:
and pivot.

ivanpopelyshev committed:
fixed destroy and tests

ivanpopelyshev committed:
Assignment by value for position/scale/skew/pivot (#2669)

* assignment by value for position/scale/skew

* and pivot.

* fixed destroy and tests

ivanpopelyshev committed:
Merge remote-tracking branch 'upstream/dev' into dev2

ivanpopelyshev committed:
fastrect fix textureid problem

themoonrat committed:
Incorrect 'new' when pushing textures into an array for static helper functions

englercj committed:
Merge pull request #2691 from themoonrat/movieclip-fix

Incorrect 'new' when pushing textures into an array for static helper functions

englercj committed:
Merge pull request #2678 from gameofbombs/dev2

FastRect problem

ivanpopelyshev committed:
fastrect mask test (#2708)

ivanpopelyshev committed:
text padding possible fix (#2709)

* fastrect mask test

* text texture fix. I didnt know that there can be negative trim. I'm sure it will affect our filters, getBounds() will be incorrect

* small fix

* Text padding fixes. works like v3

englercj commented:
Would you mind deleting this? It links to like every dev commit in the main repo and is a bit confusing :P

englercj committed:
Multiple documentation fixes

englercj committed:
Merge pull request #2710 from englercj/doc-fixes

Multiple documentation fixes

GoodBoyDigital committed:
fixed a little fast rect issue

digibo committed:
Fixed trailing slashes for xml font files' images (#2680)

andrewstart committed:
Global interaction events (#2658)

* Added touch & mouse events to InteractionManager.

* Remove whitespace only lines.

* Corrected issues with InteractionManager's global events.

rejemy committed:
Multiple ParticleContainers no longer incorrectly share properties (#1953) (#2675)

GoodBoyDigital committed:
fixes #2711

Commit:
Fixes jshint error in WebGLState

englercj committed:
Merge pull request #2716 from pixijs/fix-jshint-errors

Fixes jshint error in WebGLState

themoonrat committed:
Gradients now work with Text resolutions other than 1

themoonrat committed:
BaseTexture used for text now reports correct realWidth and realHeight (#2720)

ivanpopelyshev committed:
Wrong containsPoint of Mesh (#2718)

* fastrect mask test

* text texture fix. I didnt know that there can be negative trim. I'm sure it will affect our filters, getBounds() will be incorrect

* small fix

* Text padding fixes. works like v3

* triangle strip fix

englercj committed:
Merge pull request #2722 from themoonrat/gradient-text-resolution

Gradients now work with Text resolutions other than 1

evilangelist committed:
Fixed ParticleContainer's use of resolution in its calculations for canvas rendering (#2547)

GoodBoyDigital committed:
fixed displahyObject destroy

GoodBoyDigital committed:
Dev bounds refactor (#2729)

* getBounds refactor

* another unit test

* added get local bounds

* local bounds stuff

* bound tests and tweaks

* added Ivans recursive function back in

* updated toLocal / toGlobal

* added toLocal and toGlobal tests

* linting

* tweaked tests

* fixed text bounds

added test for it too

* test tweak for text

* more tweaks to make it pass on server

* test tweak

GoodBoyDigital committed:
republished build files

ivanpopelyshev committed:
simple transform refactoring (#2732)

* simple transform refactoring

* TransformManual deprecation

ivanpopelyshev committed:
Multitexturing DERP (#2735)

* Precision errors can be very strange on some devices. Especially if you are rendering 3d transform. This is part of gameofbombs/pixi.js fork, I didnt think that it will be needed in 2D scenes too.

* looks better

whizzkid committed:
Add NineSlicePlane.js to enable 9Slice scaling,  (#2715)

* Add NineSlicePlane.js to enable 9Slice scaling,  without the bin folder this time.

* added canvas rendering for NineSlicePlane

themoonrat committed:
Able to find out the real width and height of a base render texture, accounting for the resolution applied to it (#2741)

GoodBoyDigital committed:
fixed cacheAsBitmap filter issue

GoodBoyDigital committed:
Merge branch 'dev-transform-filter-fix' into dev

GoodBoyDigital committed:
Merge remote-tracking branch 'origin/dev' into dev

whizzkid committed:
Removed unused Mesh include (#2751)

GoodBoyDigital committed:
tweaked cacheAsBitmap

fixed issue with masks and filters + cacheAsBitmap interaction
also refactored a little to have use a CacheData that means we lower
the memory footprint of displayObjects a little.

GoodBoyDigital committed:
Merge remote-tracking branch 'origin/dev' into dev

rkrasky committed:
fix for lineHeight (#2749)

fix for lineHeight, canvas is actually too small if the line height is smaller than the font size.

rkrasky committed:
Adding performance caveat context option (#2756)

http://www.html5gamedevs.com/topic/16425-ie11-software-rendering/#comment-92760
https://msdn.microsoft.com/en-us/library/dn798645(v=vs.85).aspx

If IE or Edge shows up the message "WEBGL11258: Temporarily switching to software rendering to display webGL content", WebGL detection should fail because the software rendering is slow as hell.

neutrino84 committed:
added webgl call to setBlendMode (#2755)

GoodBoyDigital committed:
update fix for setting a matrix

GoodBoyDigital committed:
Merge remote-tracking branch 'origin/dev' into dev

GoodBoyDigital committed:
fixed cacheAsBitmap destroy bug

GoodBoyDigital committed:
republished files

GoodBoyDigital committed:
a better fix for cahceAsBitmap destroy

GoodBoyDigital committed:
bounds-tweaks (#2760)

Commit:
MaskManager - Sprite Mask Resolution (#2752)

Currently when applying a mask to a -x2 Sprite, the Sprite resolution is lowered to that of the mask.

GoodBoyDigital committed:
republished files

GoodBoyDigital committed:
renamed bounds builder to bounds

GoodBoyDigital committed:
renamed in graphics too

GoodBoyDigital committed:
republished files

rkrasky committed:
Adding CanvasRender worldAlpha (#2763)

Got an issue of the alpha value of NineSlicePlane. Adding  worldAlpha value to context if rendered via CanvasRenderer.

bigtimebuddy committed:
Adds .npmignore file (#2764)

bigtimebuddy committed:
Revert "Adds .npmignore file"

englercj committed:
Merge pull request #2766 from pixijs/revert-2764-npmignore

Revert "Adds .npmignore file"

GoodBoyDigital committed:
Delete Pixi_mini.jpg

bigtimebuddy committed:
Converts to NPM scripts, removes Gulp (#2769)

* Convert to NPM scripts, removes Gulp

* Requires node bump for jshint-stylish-summary

* Missing mkdirp

* Fixes bundling sourcemap problem

* Fixes linting

* Bumps Travis to supported NodeJS version

* Typo

* Fixes build script for Windows

* Fixes watch on Windows

* Removes unused dependency

* Remove unused dep

GoodBoyDigital committed:
same graphics object can no be rendered in multiple contexts (#2774)

same for mesh

bigtimebuddy committed:
Externalizes build process with pixify module (#2778)

GoodBoyDigital committed:
fix that allows render targets to be passed to samplers

also allowed for resolution of texture to be specified when retrieving
from the filter texture cache

GoodBoyDigital committed:
oops! missed a semicolon :)

GoodBoyDigital committed:
added pixel rounding back in

GoodBoyDigital committed:
republished files

GoodBoyDigital committed:
updated contributors

GoodBoyDigital committed:
fixes #1698

added unit test to check too

GoodBoyDigital committed:
republished files

bigtimebuddy committed:
Preprocess (#2782)

* deprecation layer is only include in debug build

* Removes the debug content

GoodBoyDigital committed:
little fix for #2697

GoodBoyDigital committed:
Merge remote-tracking branch 'origin/dev' into dev

whizzkid committed:
Dev 2679 rename params in plane (#2784)

* #2679 changed segments to vertices in param and variable names.

* #2679 better names

Commit:
Add exception handling to `canUseNewCanvasBlendModes` (#2740)

Multiple sites using pixi have been throwing the following error:
`Uncaught TypeError: Cannot read property 'data' of undefined.`

Additional context - I use a few extensions to prevent canvas fingerprinting and they seem to be the cause of my error.

bigtimebuddy committed:
Replaces Testem with Floss (#2775)

* Updates tests to use Floss/Electron

* Adds testing after building

* Don't run test twice

staff0rd committed:
fix call to getBounds (#2788)

staff0rd committed:
initialize _localID (#2789)

GoodBoyDigital committed:
sprite bounds issue

fixed issue where sprite getLocalBounds did not return correct bounds
if it had children.

GoodBoyDigital committed:
fixes for cacheAsBitmap

GoodBoyDigital committed:
fix for static cacheAsBitmap

GoodBoyDigital committed:
Merge branch 'bounds-fixes' into dev

GoodBoyDigital committed:
lint

GoodBoyDigital committed:
republished

ivanpopelyshev committed:
fastRect problem, again (#2802)

GoodBoyDigital committed:
added enabled to filters (#2798)

* added enabled to filters

* lint that code!

GoodBoyDigital committed:
A few Text updates (#2799)

* text now matchs renderers resolution

much cleaner!

* tweaked textStyle to no longer use events

its a little lighter!

* padding now applied to width as well as height

GoodBoyDigital committed:
fixed little issue with _enabledFilters being null

GoodBoyDigital committed:
republished files

GoodBoyDigital committed:
fixed textstyle emit deprecation issue

GoodBoyDigital committed:
linted :)

GoodBoyDigital committed:
double linted..

ivanpopelyshev committed:
Dev2 (#2803)

* fastRect problem, again

* remove async dependency

GoodBoyDigital committed:
republished files

GoodBoyDigital committed:
republished files

GoodBoyDigital committed:
forgot a filter check :/

GoodBoyDigital committed:
republished files

Commit:
align fix when maxWidth is used (#2780)

bigtimebuddy committed:
Adds outputName to pixify call (#2810)

bigtimebuddy committed:
Supports linting while running `npm start` (#2812)

bigtimebuddy committed:
Integrates pixify watch command (#2813)

bigtimebuddy committed:
Destroying a DisplayObject will remove itself (#2809)

* Destroying a DisplayObject will remove itself

* Container removes children before destroying

* Fixed destroy order

* Removes extra whitespace

* Updates with Ivan's suggestion for Container destroy

* Removes listeners when destroying

* More verbose destroy comments

* Typo

GoodBoyDigital committed:
tweaked the filter manager to add padding AFTER clipping it to the renderer viewport

should fix weird blur filter effect when near edges of the renderer

GoodBoyDigital committed:
republished files

ivanpopelyshev committed:
dots and commas (#2818)

GoodBoyDigital committed:
sprite bounds tweaks (#2817)

* sprite bounds tweaks

* added a trimmed sprite test

* tweaks to tint

wrong place i know..

GoodBoyDigital committed:
republished files

ivanpopelyshev committed:
jsdocs rendertarget (#2819)

GoodBoyDigital committed:
republished

staff0rd committed:
InteractionManager event documentation (#2821)

* InteractionManager event documentation

* line lenth fix

ivanpopelyshev committed:
cleaning up WebGLState (#2826)

blendMode can be 2, so we have to set 32 to avoid problems

ivanpopelyshev committed:
some stuff in renderer (#2829)

* some stuff in renderer

* unused variable

staff0rd committed:
added clarification to interactiontarget.hitArea (#2835)

rkrasky committed:
Update Text.js (#2837)

bigtimebuddy committed:
Deploy Files to CDN (#2825)

* Travis CDN deployment

* Try simple upload

* Test again

* all branches

* Add option for latest for branch

* Changed bucket names

* Specify region

* Fixed release bucket

* Remove the commits for now, only tags and branches

* all_branches also does tags

* Don't ignore bin

GoodBoyDigital committed:
added from function to tiling sprite

GoodBoyDigital committed:
republished files

bigtimebuddy committed:
Adds installation section on readme (#2811)

BrianRosamilia committed:
Fix browserify-versionify dependency (#2839)

GoodBoyDigital committed:
build publish

bigtimebuddy committed:
Adds Code of Conduct (#2844)

GoodBoyDigital committed:
made tick global in sprite renderer

fixes #2841

themoonrat committed:
Making sure the realWidth & realHeight are multiplied by the correct resolution if nothing is passed in (#2845)

GoodBoyDigital committed:
Fixes #2840 (#2847)

* Fixes #2840

Default destroy options for text are now all true.

* tweak the tests

passing unit tests..

* simpler code

thanks @bigtimebuddy !

GoodBoyDigital committed:
republished files

GoodBoyDigital committed:
Merge branch 'dev'

# Conflicts:
#	README.md
#	bin/pixi.js
#	bin/pixi.js.map
#	bin/pixi.min.js
#	bin/pixi.min.js.map
#	package.json
#	src/accessibility/AccessibilityManager.js
#	src/core/display/Container.js
#	src/core/renderers/webgl/WebGLRenderer.js
#	src/core/text/Text.js
#	src/extras/TilingSprite.js
#	src/interaction/InteractionManager.js
#	src/loaders/spritesheetParser.js

Commit:
Merge branch 'master' into dev

bigtimebuddy committed:
Updated docs example with fontFamily and fontSize (#2861)

whizzkid committed:
Fix process interactive (#2855)

* Added 1 extra check in the procesInteractive to check if children exists before attempting to call children.length
this would've thrown an error if you remove stuff in an onMouseClick for example.

* smaller diff maybe

* changed remark

* Better and faster fix
set interactive and interactiveChildren to false on the DisplayObject.destroy()
this way no extra check is needed in the processInteractive function.

bigtimebuddy committed:
Removes link (breaks docs), updates year (#2857)

bigtimebuddy committed:
Fixes a typo and the link for CDN (#2860)

* Fixes a typo and the link for CDN

* Fixes link

GoodBoyDigital committed:
fixed incorrect height setter for text

also added a unit test

GoodBoyDigital committed:
fixes #2871

Text width was being incorrectly reported when padding is used

GoodBoyDigital committed:
Update README.md

GoodBoyDigital committed:
fixed the jump blur effect

ivanpopelyshev committed:
Dev2 (#2880)

* Update README.md

* fastrect sprite tint+color

GoodBoyDigital committed:
fixes #2873

GoodBoyDigital committed:
Merge remote-tracking branch 'origin/dev' into dev

GoodBoyDigital committed:
removed use of png data in test canvas blend modes

GoodBoyDigital committed:
republished files

bigtimebuddy committed:
Support for built, deployable documentation (#2864)

* Add docs to branch

* Adds multiple bin and docs

* Bump jaguarjs

bigtimebuddy committed:
Removes deprecation logs for release (#2887)

GoodBoyDigital committed:
fixes #2884

GoodBoyDigital committed:
tweak to from function

if theres nothing left - assume the source is a texture

GoodBoyDigital committed:
fixes #2883

GoodBoyDigital committed:
Update README.md

marooou committed:
Fix invalid hit-testing within containers cached as bitmaps. [#2895] (#2896)

GoodBoyDigital committed:
enable texture gc by default

tresf committed:
Fix gallery link (#2912)

GoodBoyDigital committed:
fixes #2910

GoodBoyDigital committed:
republished files

GoodBoyDigital committed:
fixes #2915

GoodBoyDigital committed:
rebuilt

GoodBoyDigital committed:
Merge branch 'dev'

GoodBoyDigital committed:
updated version

GoodBoyDigital committed:
republished files with build increment

GoodBoyDigital committed:
republished files

GoodBoyDigital committed:
Merge branch 'dev'

GoodBoyDigital committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	README.md

GoodBoyDigital committed:
add attributes to shader

GoodBoyDigital committed:
added left/right/bottom/top props to rect

GoodBoyDigital committed:
Update README.md

GoodBoyDigital committed:
Update LICENSE

GoodBoyDigital committed:
Update README.md

themoonrat committed:
Couple of changes to allow cocoon js to play nice with pixi
Do not create touch hooks for accessibility - this stops cocoon from working
Do not use gradient text, just use the first colour listed in the array - cocoon displays text with gradients as black otherwise

leonardo-silva committed:
Implementing classes using the new ES6 syntax (#2936)

* Migration to ES6 classes

* Migration to ES6 classes

* Migration to ES6 classes

* Migration to ES6 classes

* Using babelify to transpile ES6 code

* Adding babel-preset-es2015 in devDependencies

Commit:
Disallow the use of `var`

Commit:
Update imports and exports, getters and setters

CodeRect committed:
small fixes in the documentation

CodeRect committed:
small fixes in the documentation

bigtimebuddy committed:
Replaces jshint with eslint, using babel in loose mode

GoodBoyDigital committed:
Merge branch 'master' into dev

Commit:
Remove reporter

GoodBoyDigital committed:
fix filter offset issue

GoodBoyDigital committed:
Small updates to fix some es6 conversion issues

- Removed some require references that were preventing mixins like
interaction from working.
- Fixed Text

bigtimebuddy committed:
Merge pull request #2960 from pixijs/dev-es6-updates

Small updates to fix some es6 conversion issues

ivanpopelyshev committed:
Multiple renerers for Graphics fastRect, solution that will work for sure.

GoodBoyDigital committed:
fixed filter issue

ivanpopelyshev committed:
Merge pull request #2963 from gameofbombs/dev4

Multiple renerers for Graphics fastRect, solution that will work for sure

ivanpopelyshev committed:
Merge pull request #2964 from pixijs/master

Derp merge master-dev

bigtimebuddy committed:
Merge pull request #2962 from pixijs/dev-es6-updates

fixed filter issue

bigtimebuddy committed:
typo

stbaer committed:
removed old jshint comments; replaced jshint in CONTRIBUTING.md with eslint

bigtimebuddy committed:
Merge pull request #2973 from stbaer/dev-es6-jshint-cleanup

removed old jshint comments; replaced jshint in CONTRIBUTING.md with …

bigtimebuddy committed:
Merged with 'dev'

# Conflicts:
#	src/core/Shader.js
#	src/core/graphics/Graphics.js
#	src/core/math/shapes/Rectangle.js
#	src/core/renderers/webgl/managers/FilterManager.js
#	src/core/sprites/webgl/generateMultiTextureShader.js

bigtimebuddy committed:
Merge remote-tracking branch 'origin/dev-es6' into dev-es6

bigtimebuddy committed:
Fixes lint error

Commit:
Fixed Ticker's internal _tick method

stbaer committed:
some more es6 changes (#2969)

* some more es6 changes
- const instead of let if possible
- arrow functions
- string templates
- argument defaults

* object property shorthand

* some more es6 changes
- const instead of let if possible
- arrow functions
- string templates
- argument defaults

* Revert "Merge branch 'dev-es6-sb' of github.com:stbaer/pixi.js into dev-es6-sb"

This reverts commit 8d3af819529c9899df0302c254d730dd3ebfdef6, reversing
changes made to e8b994803c9faabe104c177cbf72c68404fef53c.

* Revert "Merge branch 'dev-es6' into dev-es6-sb"

This reverts commit 3eeaf79220a65a3fde644f30d21dab4b288bce33, reversing
changes made to fb695360b0b54a7024d079b305ebb351833c6914.

* Revert "Revert "Merge branch 'dev-es6-sb' of github.com:stbaer/pixi.js into dev-es6-sb""

This reverts commit c860228b67cd3a99e35ba37654667d652bcede94.

* fix RenderTexture, incomplete merge

* fix for text font properties

* reverted some changes

* checked out License, Readme and Contributinf from dev-es6

* editorconfig updated: 2 spaces for .eslintrc

* added eslint rule: spaces around fat arrow

* used eslint fix for consistent spaces around fat arrow

* don't allow scale.x or scale.y to be 0 in displayObject setTransform

* removed jshint comments (again)

englercj committed:
Merge pull request #2946 from themoonrat/cocoon-fixes

Fixes for PIXI + Cocoon

bigtimebuddy committed:
Merge branch 'dev' into dev-es6

# Conflicts:
#	src/accessibility/AccessibilityManager.js
#	src/core/text/Text.js

GoodBoyDigital committed:
Update LICENSE

GoodBoyDigital committed:
webpack temp fix (#2981)

* webpack temp fix

* use minified version

GoodBoyDigital committed:
version update

bigtimebuddy committed:
Merge branch 'dev' into dev-es6

themoonrat committed:
Set attributes to play videos without going full screen on iOS10 (#2982)

themoonrat committed:
Jsdoc fixes - marking relevant functions as @private (#2978)

* Adding private jsdoc tag

* Added jsdoc @private tag to relevant functions

* moved private variables to above the setter, but out of the way of getting jsdocced

englercj committed:
Update Contributing Guide with new branch flow. (#2983)

* Update Contributing Guide with new branch flow.

Added more details, links to GitHub docs about forking, and the new
branching flow. As a bonus, also changed some minor things in the
editorconfig.

* Fix typo.

* a couple more minor details

GoodBoyDigital committed:
Merge branch 'dev-es6' into dev

# Conflicts:
#	.editorconfig
#	CONTRIBUTING.md
#	src/core/display/DisplayObject.js
#	src/core/graphics/Graphics.js
#	src/core/renderers/webgl/WebGLRenderer.js
#	src/core/sprites/Sprite.js
#	src/core/text/Text.js
#	src/core/textures/VideoBaseTexture.js
#	src/filters/colormatrix/ColorMatrixFilter.js
#	src/mesh/Mesh.js

GoodBoyDigital committed:
added rectangle bound checks

GoodBoyDigital committed:
added inline play

GoodBoyDigital committed:
fast rect fix

GoodBoyDigital committed:
filter pad fix

GoodBoyDigital committed:
republished files

bigtimebuddy committed:
Create Bower workflow which removes bin files

bigtimebuddy committed:
Space

bigtimebuddy committed:
Don't build twice

englercj committed:
add a note about using modern node

marekventur committed:
Update documentation for TextStyle.padding

The change has been made here: https://github.com/pixijs/pixi.js/commit/52d86a2bae9631142d4635e8421d2761023cbcbc

bigtimebuddy committed:
Organizes gitignore

englercj committed:
Merge pull request #2989 from marekventur/patch-2

Update documentation for TextStyle.padding

englercj committed:
Manually merge #2970

englercj committed:
Merge branch 'CodeRect-patch-1' into dev

themoonrat committed:
Fixes mouseover behaviour

englercj committed:
Manually merge #2929

englercj committed:
Merge pull request #2996 from themoonrat/interaction-mouseover-fix-es6

Fixes mouseover behaviour

englercj committed:
Manually merge #2925

ivanpopelyshev committed:
bounds and transform for invisible containers.

endel committed:
add InteractionEvent class. assigning event's target/currentTarget.

stbaer committed:
fixes autoDetectRenderer in core/index.js

englercj committed:
add some missed doc changed from #2954

stbaer committed:
graphics lineStyle default args and jsdoc

stbaer committed:
jsdoc

bigtimebuddy committed:
Detached test, install and build scripts

englercj committed:
Merge pull request #3003 from stbaer/dev-graphics-linestyle

graphics lineStyle default args and jsdoc

stbaer committed:
Prefer const (#3002)

* added prefer-const rule to .eslintrc

* eslint const autofixes

* manual let to const changes; lint free;

* added brace style 'allman' to eslintrc

* brace style, comment fixes and default arg

* brace style fixes

* more brace styles

* Point default args

* Point brace styles

* Observable Point default args

* more brace styles

* jsdoc comment style

* brace style

* brace styles and small jsdoc fix

* brace styles

* removed unneeded eslint inline comment

* brace styles; lint free

stbaer committed:
small jsdoc fix - added some missing (#3008)

englercj committed:
Merge pull request #3000 from stbaer/dev-pr

fixes autoDetectRenderer in core/index.js

RanzQ committed:
Better SVG support (#2614)

* Better SVG support

* Images are loaded using canvas which enables pre-scaled textures
* Scaling is done by providing an extra parameter `sourceScale` to the `fromImage` method
* Size of the original SVG source is checked with an extra XHR (necessary step for the canvas API to work)
* Supports data URI based SVG

* Fixed brace-style.

* Added util tests.

* Rebased to latest dev. Added 'px' support.

themoonrat committed:
Adding support for pointer events (#2997)

* Adding support for pointer events

* eslint fix

* more eslint

* And again, really need an npm command somewhere to do the auto eslint fix with the correct spacings

* eslint take #12

englercj committed:
Merge pull request #2986 from pixijs/bower-workflow

Improve Bower Workflow

themoonrat committed:
CanvasPrepare updated to simulate WebGLPrepare (#2959)

* For WebGLPrepare- the done callback is now optional
For CanvasPrepare - an implimentation has been added rathar than it just being a stub. Uploading textures to the GPU can be simulated by forcing a draw command onto a hidden using these textures.

* Fixed couple of context issues

themoonrat committed:
Allows a callback to be set on when the frame changes on a MovieClip (#2948)

* Allows a callback to be set on whenever the frame changes on a movieclip

* currentFrame passed through the onFrameChange callback

englercj committed:
fix canvas resolution; manually merge #2870

englercj committed:
fix issues w/ Container destroy; see #2863 and #2995

englercj committed:
merge #2828

englercj committed:
Merge branch 'endel-interaction-targets' into dev

englercj committed:
merge #2956

englercj committed:
Merge branch 'gameofbombs-dev2' into dev

themoonrat committed:
Fixes to have pointer events work with the new eventData class

themoonrat committed:
Always clear Text Canvas via clearRect

peteward44 committed:
Fix issue when creating a BaseTexture with a 'null' source value, then using .loadSource() with an already-loaded IMG element. The 'loaded' event is never fired in that case and Texture objects are forever waiting for the 'loaded' event to be emitted

englercj committed:
Merge pull request #3015 from themoonrat/clearing-text-canvas

Always clear Text Canvas via clearRect

englercj committed:
Merge pull request #3016 from peteward44/dev

Fixed issue with delay-loaded BaseTexture objects

englercj committed:
Merge pull request #3014 from themoonrat/interaction-target-for-pointer-events

Fixes for pointer to work with new InteractionEvent class

Commit:
Fix linting issues

englercj committed:
Merge pull request #3018 from pixijs/fix-linting

Fix linting issues
